tests/unit/amo/components/TestSearchForm.js

```diff
@@ -40,140 +39,226 @@ describe('<SearchForm />', () => {
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
 
-  it('passes the context through', () => {
-    const { store } = dispatchSignInActions();
-    store.dispatch(setViewContext(ADDON_TYPE_EXTENSION));
+      // setting the `query` prop to empty also sets the input state to empty.
+      const wrapper = mount(
+        <SearchFormWrapper
+          query=""
+          suggestions={autocompleteState.suggestions}
+        />
+      );
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+    });
 
-    const state = store.getState();
+    it('does not display suggestions when there is no suggestion', () => {
+      const wrapper = mount(
+        <SearchFormWrapper
+          suggestions={[]}
+        />
+      );
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+    });
 
-    expect(mapStateToProps(state).addonType).toEqual(ADDON_TYPE_EXTENSION);
+    it('displays 10 loading bars when suggestions are loading', () => {
+      const wrapper = mount(
+        <SearchFormWrapper
+          suggestions={[]}
+          loadingSuggestions
+        />
+      );
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      expect(wrapper.find(LoadingText)).toHaveLength(10);
+    });
   });
 
-  it('does not set addonType if context is not a validAddonType', () => {
-    const { store } = dispatchSignInActions();
-    store.dispatch(setViewContext(VIEW_CONTEXT_HOME));
+  describe('mapStateToProps', () => {
+    it('passes the api through', () => {
+      const { store } = dispatchSignInActions();
 
-    const state = store.getState();
+      const state = store.getState();
 
-    expect(mapStateToProps(state).addonType).toEqual(null);
+      expect(mapStateToProps(state).api).toEqual(state.api);
+    });
+
+    it('passes the context through', () => {
+      const { store } = dispatchSignInActions();
+      store.dispatch(setViewContext(ADDON_TYPE_EXTENSION));
+
+      const state = store.getState();
+
+      expect(mapStateToProps(state).addonType).toEqual(ADDON_TYPE_EXTENSION);
+    });
+
+    it('does not set addonType if context is not a validAddonType', () => {
+      const { store } = dispatchSignInActions();
+      store.dispatch(setViewContext(VIEW_CONTEXT_HOME));
+
+      const state = store.getState();
+
+      expect(mapStateToProps(state).addonType).toEqual(null);
+    });
+
+    it('passes the suggestions through', () => {
+      const result = createFakeAutocompleteResult();
+      const { store } = dispatchAutocompleteResults({ results: [result] });
+
+      const state = store.getState();
+
+      expect(mapStateToProps(state).suggestions).toEqual([
+        {
+          iconUrl: result.icon_url,
+          name: result.name,
+          url: result.url,
+        },
+      ]);
+      expect(mapStateToProps(state).loadingSuggestions).toBe(false);
```

Comment:
It would be nice to have another test that asserts `loadingSuggestions` gets set to true. You could set it up with a dispatch like:

```js
store.dispatch(autocompleteStart(...));
```